sklearn_numba_dpex/common/kernels.py

```diff
@@ -216,46 +216,70 @@ def make_sum_reduction_2d_kernel(
     sub_group_size=None,
     fused_elementwise_func=None,
 ):
-    """Implement data_2d.sum(axis=axis) or data_1d.sum().
+    """Compute data_2d.sum(axis=axis) or data_1d.sum().
 
     This implementation is optimized for C-contiguous arrays.
 
-    numba_dpex does not provide tools such as `cuda.reduce` so we implement from scratch
-    a reduction strategy. The strategy relies on the commutativity of the operation used
-    for the reduction, thus allowing to reduce the input in any order.
-
-    The strategy consists in performing local reductions in each work group using local
-    memory where each work item combine two values, thus halving the number of values,
-    and the number of active work items. At each iteration the work items are discarded
-    in a bracket manner. The work items with the greatest ids are discarded first, and
-    we rely on the fact that the remaining work items are adjacents to optimize the RW
-    operations.
-
-    Once the reduction is done in a work group the result is written in global memory,
-    thus creating an intermediary result whose size is divided by
-    `2 * work_group_size` if axis=1, or by `2 * work_group_size // sub_group_size` if
-    axis=0. This is repeated as many times as needed until only one value remains in
-    global memory.
-
-    If `fused_elementwise_func` is not None, it will be applied element-wise before
-    summing. It must be a function that will be interpreted as a dpex.func and is
-    subject to the same rules ( see
+    numba_dpex does not provide tools such as `cuda.reduce` so we implement
```

Comment:
In scikit-learn learn we still use 79 char limits for the docstrings (while we use 88 char limit for the code itself).

I did not reconfigure my editor when making those edit so I used the scikit-learn convention here.

Note that strictly following PEP8 would require 72 chars for the docstrings but I think nowadays few projects follow this.

But since this is not scikit-learn, I don't have a strong opinion, feel free to rewrap to 88.

Reply:
It's more that I like to have automated tools that enforce things to avoid future discussions in PRs. Is it something that is enforced by the CI in scikit-learn ?